canonicalwebteam/image_template/__init__.py

```diff
@@ -42,19 +42,23 @@ def image_template(
     hi_def_cloudinary_options = cloudinary_options.copy()
 
     std_def_cloudinary_options.append("w_" + str(width))
-    std_def_cloudinary_options.append("h_" + str(height))
+
+    if height != "auto":
```

Comment:
Does the name "auto" actually carry any meaning at any point? You're not passing it through to Cloudinary, and it's not valid in the HTML `<img>` tag (even though you're putting it there):

> The intrinsic height of the image, in pixels. **Must be an integer without a unit.**
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attr-height

So I think we should:

- Change the `image_template` definition on line 15 to make height optional by setting `height=None`
- If `height` isn't set, don't pass it through to the template
- Update the template to exclude the `height=""` parameter if `height` isn't set
- Update the README.md to explain that height is optional, although I'd like to still ask people to *please* set height if they possibly can, to avoid the jumping issue (and link to something [like this](https://itnext.io/how-to-stop-content-jumping-when-images-load-7c915e47f576) to provide more flavour).

Reply:
You could maybe link to the discussion in https://github.com/canonical-web-and-design/canonicalwebteam.image-template/issues/38 too. We should probably also update that issues to mention this change.

Reply:
If I am to make `height` optional, in order to preserve the order of the parameters in the function, because `hi_def` is at the moment mandatory, would it be an acceptable option to set `hi_def` default to `True`? @nottrobin

Reply:
I don't think there's any particular need to preserve the order. If `hi_def` is supposed to be mandatory then just move it higher up in the parameters list.

Reply:
Changing the order of the parameters would force us to update every single place `image_template` was ever used on any of our websites. Wouldn't it?

Reply:
I don't think this will be so hard. In any case where named arguments are used (`width=10, height=20`) rather than positional (`10, 20`) won't need to change. I also think basically everywhere does it through the template tag, which I think basically enforces named arguments, so probably wouldn't need to change. Worth checking though.

Anyway, in every case projects will explicitly update to the new version as and when they get around to it. If the upgrade requires a bit of reworking it's not so bad. The projects can just stay on the old version until someone has time.